image/internal/manifest/manifest_test.go

```diff
@@ -112,6 +116,74 @@ func TestMatchesDigest(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestMatchesDigestWithSHA512(t *testing.T) {
```

Comment:
Nah. It doesn’t work for a caller with a single sha512 digest do change a process-wide global (racing against other goroutines), _and_ it’s entirely unnecessary.

… and, after that is fixed, this should be tested just by a few more table entries in the existing test.